utility/drivers.cpp

```diff
@@ -747,54 +784,27 @@ boolean ml8511_init()
  */
 float ml8511_getUV(int pin)
 {
-  int uv_v = average_analog_read(pin);
-  int ref_v = average_analog_read(DRIVER_ML8511_REF_PIN);
-  float scaled_uv_v = 3.3 / ref_v * uv_v;
+  float scaled_uv_v;
+  if (ARDUSAT_SPACEBOARD) {
+    scaled_uv_v = ml8511_uv_adc.read_uv();
+  } else {
+    int uv_v = average_analog_read(pin);
+    int ref_v = average_analog_read(DRIVER_ML8511_REF_PIN);
+    scaled_uv_v = 3.3 / ref_v * uv_v;
+  }
   return map_float(scaled_uv_v, 0.99, 2.9, 0.0, 15.0);
 }
 
 
 /*
- * SI1145 UV/Light
- */
-Adafruit_SI1145 si1145_uv = Adafruit_SI1145();
-
-boolean si1145_init() {
-  return si1145_uv.begin();
-}
-
-float si1145_getUVIndex() {
-  float UVindex = si1145_uv.readUV();
-
-  // the index is multiplied by 100 so to get the integer index, divide by 100
-  UVindex /= 100.0;
-  return (UVindex);
-}
-
-/*
  * MLX90614 IR Temperature
  */
-uint16_t _mlx90614_read16(uint8_t a) {
-  uint16_t ret;
-
-  Wire.beginTransmission(DRIVER_MLX90614_ADDR); // start transmission to device
-  Wire.write(a); // sends register address to read from
-  Wire.endTransmission(false); // end transmission
-
-  Wire.requestFrom((uint8_t) DRIVER_MLX90614_ADDR, (uint8_t)3);// send data n-bytes read
-  ret = Wire.read(); // receive DATA
-  ret |= Wire.read() << 8; // receive DATA
-
-  uint8_t pec = Wire.read();
-
-  return ret;
-}
-
 float _mlx90614_readTemp(uint8_t reg) {
-  float temp;
+  uint32_t data;
+
+  readFromRegAddr(DRIVER_MLX90614_ADDR, reg, &data, 3, LITTLE_ENDIAN);
+  float temp = (data & 0xFFFF);
```

Comment:
I'm honestly not sure. I just ported over the logic from lines 784-788 to use the `readFromRegAddr` utility function. I looked for some documentation but couldn't find any, because when I had it read only two bytes I got bad ir temperature values. Notice line 788: `uint8_t pec = Wire.read();`, it was reading an extra byte that was never used? I didn't investigate this much.